provider_client.go

```diff
@@ -335,6 +351,29 @@ func (client *ProviderClient) Request(method, url string, options *RequestOpts)
 	})
 }
 
+func (client *ProviderClient) retryBackoff(respErr *ErrUnexpectedResponseCode, e error, retries uint) error {
+	var sleep time.Duration
+
+	retryAfter := respErr.ResponseHeader.Get("Retry-After")
+	if retryAfter == "" {
+		return e
+	}
+
+	if v, err := strconv.ParseUint(retryAfter, 10, 32); err == nil {
+		sleep = time.Duration(v) * time.Second
+	}
+
+	if v, err := time.Parse(http.TimeFormat, retryAfter); err != nil {
```

Comment:
Should this code only execute if the above `ParseUint` was unsuccessful?

Reply:
thanks, fixed